x-pack/legacy/plugins/ml/public/data_frame/pages/transform_management/services/transform_service/delete_transform.ts

```diff
@@ -15,26 +15,39 @@ import {
   DataFrameTransformListRow,
 } from '../../components/transform_list/common';
 
-export const deleteTransform = async (d: DataFrameTransformListRow) => {
-  try {
-    if (d.stats.state === DATA_FRAME_TRANSFORM_STATE.FAILED) {
-      await ml.dataFrame.stopDataFrameTransform(d.config.id, true, true);
+export const deleteTransforms = async (dataFrames: DataFrameTransformListRow[]) => {
+  const results = await Promise.all(
+    dataFrames.map(async df => {
```

Comment:
closing and deleting asynchronously from the browser is going to put a lot of strain on the kibana and es server.
This has been intentionally avoided in the anomaly detector job management page. The bulk delete/close actions are behind a single kibana endpoint which then loops over the list and performs a force delete/close synchronously.
It depends on whether the transform delete/close action is expensive, it's probably not as bad as the anomaly detector jobs, but I think these bulk actions should probably be performed in the the same way.

Reply:
Great point. Does it sound like a good idea to add as a follow up change/refactor?
cc @walterra, @droberts195  (for context)

Reply:
IMHO in this PR let's at least change the behaviour to not fire all requests at once using `Promise.all` but do it in sequence (also for the other actions). This would fire up to 50 requests in parallel if a user would bulk delete all transforms on a list with paging set to 50.

Reply:
After a chat with @walterra - I'll go ahead and make the change in this PR to move the bulk actions behind a single Kibana endpoint to loop over the list of ids and perform the action synchronously.